clang/include/clang/Sema/Sema.h

```diff
@@ -1384,6 +1384,9 @@ class Sema {
   /// Retrieve the module loader associated with the preprocessor.
   ModuleLoader &getModuleLoader() const;
 
+  /// Invent a new identifier for constrained template parameters.
+  IdentifierInfo *InventIdentifier(StringRef Name, unsigned Index);
```

Comment:
InventAbbreviatedTemplateParameterTypeName

Reply:
Also rename first parameter to "ParamName" and accept an IdentifierInfo * (check if it's null for an unnamed parameter)